featurebyte/routes/feature_store/controller.py

```diff
@@ -96,52 +100,51 @@ async def create_feature_store(
+            # Check credentials
+            credential = CredentialModel(
+                name=document.name,
+                feature_store_id=document.id,
+                database_credential=data.database_credential,
+                storage_credential=data.storage_credential,
+            )
 
-        async def _updated_get_credential(user_id: str, feature_store_name: str) -> Any:
-            """
-            Updated get_credential will try to look up the credentials from config.
-
-            If there are credentials in the config, we will ignore whatever is passed in here.
-            If not, we will use the params that are passed in.
-
-            Parameters
-            ----------
-            user_id: str
-                user id
-            feature_store_name: str
-                feature store name
-
-            Returns
-            -------
-            Any
-                credentials
-            """
-            cred = await get_credential(user_id, feature_store_name)
-            if cred is not None:
-                return cred
-            return credential
-
-        get_credential_to_use = _updated_get_credential
-        await self.session_validator_service.validate_feature_store_id_not_used_in_warehouse(
-            feature_store_name=data.name,
-            session_type=data.type,
-            details=data.details,
-            get_credential=get_credential_to_use,
-            users_feature_store_id=document.id,
-        )
-        logger.debug("End validate_feature_store_id_not_used_in_warehouse")
+            async def _updated_get_credential(user_id: str, feature_store_name: str) -> Any:
+                """
+                Updated get_credential will try to look up the credentials from config.
+
+                If there are credentials in the config, we will ignore whatever is passed in here.
+                If not, we will use the params that are passed in.
+
+                Parameters
+                ----------
+                user_id: str
+                    user id
+                feature_store_name: str
+                    feature store name
+
+                Returns
+                -------
+                Any
+                    credentials
+                """
+                cred = await get_credential(user_id, feature_store_name)
+                if cred is not None:
+                    return cred
+                return credential
+
+            get_credential_to_use = _updated_get_credential
+            await self.session_validator_service.validate_feature_store_id_not_used_in_warehouse(
+                feature_store_name=data.name,
+                session_type=data.type,
+                details=data.details,
+                get_credential=get_credential_to_use,
+                users_feature_store_id=document.id,
+            )
+            logger.debug("End validate_feature_store_id_not_used_in_warehouse")
 
-        async with self.service.persistent.start_transaction():
```

Comment:
@jevonyeoh Replaced transaction with try / catch + clean up here as a transaction times out in 60s. For slow connections this can sometimes be an issue.

Reply:
sounds good - any reason not to place the `try` line here instead where the txn previously was? that way the lines above won't have to be changed, and the scope of the `try-catch` block is as narrow as possible